lib/puppet/file_serving/metadata.rb

```diff
@@ -121,23 +138,30 @@ def initialize(path,data={})
     @checksum_type ||= Puppet[:digest_algorithm]
     @ftype       = data.delete('type')
     @destination = data.delete('destination')
+    @inlinable   = data.delete('inlinable')
     super(path,data)
   end
 
   def to_data_hash
-    super.update(
-      {
-        'owner'        => owner,
-        'group'        => group,
-        'mode'         => mode,
-        'checksum'     => {
-          'type'   => checksum_type,
-          'value'  => checksum
-        },
-        'type'         => ftype,
-        'destination'  => destination,
-      }
-    )
+    data_hash = {
+      'owner'        => owner,
+      'group'        => group,
+      'mode'         => mode,
+      'checksum'     => {
+        'type'   => checksum_type,
+        'value'  => checksum
+      },
+      'type'         => ftype,
+      'destination'  => destination,
+    }
+
+    # To maintain wire compatibility, we don't add the inlinable
+    # attribute to the serialized format unless it was explicitly set.
+    if !inlinable.nil?
```

Comment:
I've got these little checks in a few places around file metadata, trying to ensure that in cases where agent and server versions don't match, we don't change the wire format of these structs. I don't really know how strict Puppet's validation is or if this is a worthwhile endeavor